controller/scan/scan.go

```diff
@@ -235,14 +235,18 @@ func AddScanner(id string) {
 		regScher.Init()
 	}
 
-	regScher.AddProcessor(id)
+	if err := regScher.AddProcessor(id); err != nil {
```

Comment:
What would be the side effects if those actions failed? Should we populate the error to the caller?

Reply:
Hi @yasker, good question! When this happens, it means someone tries to add a processor when it's already added. It's a programming error and should be fixed by the developer. The best action here is to log the error so the developer knows.

Reply:
Similar to EINVAL errno.

Reply:
If it's a duplicate provider, it should be fine. However, there is no error type check here. It's not immediately obvious it will only due to a duplicate provider by reading the code. The normal process is to populate the error up in the chain to the caller. Should we populate the error here as well?

Reply:
yes I think that would be perfect if we can have all the code follow this convention, which is golang's standard.  I can work on it.  
Currently we don't have that kind of coding convention, which makes error handling and logging very complicated.  I think we can discuss about this in the meeting next week too.

Reply:
Let's raise it up next week. We need to consider the error path to improve the robustness of the code.